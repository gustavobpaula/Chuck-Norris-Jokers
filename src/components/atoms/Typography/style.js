import styled, { css } from 'styled-components';
import { pxToRem } from '../../../utils';
import designSystem from '../../../designSystem';

const variants = {
  logo: () => css`
    font-size: ${pxToRem(50)};

    @media (max-width: ${designSystem.get('display.sizes.phone')}) {
      font-size: ${pxToRem(25)};
    }
  `,

  joke: () => css`
    font-size: ${pxToRem(60)};
    text-shadow: -1px 0 ${designSystem.get('colors.color3')},
      0 1px ${designSystem.get('colors.color3')},
      1px 0 ${designSystem.get('colors.color3')},
      0 -1px ${designSystem.get('colors.color3')};

    @media (max-width: ${designSystem.get('display.sizes.phone')}) {
      font-size: ${pxToRem(30)};
    }
  `,

  category: () => css`
    font-size: ${pxToRem(40)};

    @media (max-width: ${designSystem.get('display.sizes.phone')}) {
      font-size: ${pxToRem(22)};
    }
  `,

  button: () => css`
    font-size: ${pxToRem(30)};

    @media (max-width: ${designSystem.get('display.sizes.phone')}) {
      font-size: ${pxToRem(22)};
    }
  `,
};

export const Typography = styled.span`
  ${props => variants[props.variant]}
`;
