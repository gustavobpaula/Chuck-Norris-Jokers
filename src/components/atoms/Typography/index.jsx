import React from 'react';
import PropTypes from 'prop-types';
import * as Styled from './style';

const Typography = ({ component, variant, children, ...rest }) => (
  <Styled.Typography as={component} variant={variant} {...rest}>
    {children}
  </Styled.Typography>
);

Typography.defaultProps = {
  variant: 'paragraph',
  component: 'span',
};

Typography.propTypes = {
  component: PropTypes.string,
  children: PropTypes.node.isRequired,
  variant: PropTypes.string,
};

export default Typography;
