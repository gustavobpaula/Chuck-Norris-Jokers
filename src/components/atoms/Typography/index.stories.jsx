import React from 'react';
import Typography from './index';

export default {
  component: Typography,
  title: 'Atoms/Typography',
};

export const element = () => (
  <Typography component="h1">Component Typography</Typography>
);
element.story = {
  name: 'Default',
};

export const logo = () => (
  <Typography variant="logo" component="h2">
    Logo Variant
  </Typography>
);
logo.story = {
  name: 'Logo Variant',
};

export const joke = () => (
  <Typography variant="joke" component="h2">
    Joke Variant
  </Typography>
);
joke.story = {
  name: 'Joke Variant',
};

export const category = () => (
  <Typography variant="category" component="h2">
    Category Variant
  </Typography>
);
category.story = {
  name: 'Category Variant',
};

export const button = () => (
  <Typography variant="button" component="h2">
    Button Variant
  </Typography>
);
button.story = {
  name: 'Button Variant',
};
