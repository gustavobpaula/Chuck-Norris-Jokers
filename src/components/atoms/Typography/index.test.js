import React from 'react';
import { mount } from 'enzyme';
import Typography from './index';

describe('<Typography />', () => {
  it('should mount properly', async () => {
    const wrapper = mount(<Typography>Component Typography</Typography>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should has a h1 tag', async () => {
    const wrapper = mount(
      <Typography component="h1">Component Typography</Typography>
    );

    expect(wrapper.find('h1')).toMatchSnapshot();
  });
});
