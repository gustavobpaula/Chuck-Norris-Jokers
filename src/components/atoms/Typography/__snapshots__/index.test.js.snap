// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Typography /> should has a h1 tag 1`] = `
<h1
  className="style__Typography-sc-8vyb1l-0 cBZQFp"
>
  Component Typography
</h1>
`;

exports[`<Typography /> should mount properly 1`] = `
<Typography
  component="span"
  variant="paragraph"
>
  <style__Typography
    as="span"
    variant="paragraph"
  >
    <StyledComponent
      as="span"
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "style__Typography-sc-8vyb1l-0",
            "isStatic": false,
            "lastClassName": "cBZQFp",
            "rules": Array [
              [Function],
            ],
          },
          "displayName": "style__Typography",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "style__Typography-sc-8vyb1l-0",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      variant="paragraph"
    >
      <span
        className="style__Typography-sc-8vyb1l-0 cBZQFp"
      >
        Component Typography
      </span>
    </StyledComponent>
  </style__Typography>
</Typography>
`;
